src/features/components/templates/trade-platform/how-to-start/index.tsx

```diff
@@ -21,29 +21,23 @@ const TradingPlatformHowToStart = ({
 }: TradingPlatformStartProps) => {
     const [tab, setTab] = useState('demo')
     const { is_mobile_or_tablet } = useBreakpoints()
-    const default_item = is_mobile_or_tablet ? demo_mobile : demo
-    const [items, setItems] = useState<TabItemTypes[]>(default_item)
+    const [items, setItems] = useState<TabItemTypes[]>(is_mobile_or_tablet ? demo_mobile : demo)
 
-    const onTabClick = useCallback(
-        (chosen_tab: TabType) => {
-            setTab(chosen_tab)
+    const onTabClick = (chosen_tab: TabType) => {
+        setTab(chosen_tab)
+    }
+    const new_items = useMemo(() => {
+        switch (tab) {
+            case 'real':
+                return is_mobile_or_tablet ? real_mobile : real
+            case 'demo':
+                return is_mobile_or_tablet ? demo_mobile : demo
+        }
+    }, [demo, demo_mobile, is_mobile_or_tablet, real, real_mobile, tab])
```

Comment:
```suggestion
    const onTabClick = (chosen_tab: TabType) => {
        setTab(chosen_tab)
        switch (chosen_tab) {
            case 'real':
                is_mobile_or_tablet ? setItems(real_mobile): setItems(real):
            case 'demo':
                is_mobile_or_tablet ? setItems(demo_mobile): setItems(demo)
        }
    }
```

Reply:
Can't we use it this way? Then we don't have to use extra `useMemo` and `useEffect`

Reply:
hi bro,  this function has dependency with  (demo, demo_mobile, is_mobile_or_tablet, real, real_mobile, tab)
these variables, we need to re-calculate items when tab change , mobile view change so we have some dependency 
so we need to use a hook 
but using use effect will cost some additional rerendering so to avoid that I have used use memo to avoid additional re-rendering and memoize value and optimize performance by caching.